components/script/dom/xmlhttprequest.rs

```diff
@@ -1377,11 +1383,33 @@ impl Extractable for BodyInit {
                 let bytes = b.get_bytes().unwrap_or(vec![]);
                 (bytes, content_type)
             }
-            BodyInit::FormData(ref formdata) => {
+            DocumentOrBodyInit::FormData(ref formdata) => {
                 let boundary = generate_boundary();
                 let bytes = encode_multipart_form_data(&mut formdata.datums(), boundary.clone(),
                                                        UTF_8 as EncodingRef);
                 (bytes, Some(DOMString::from(format!("multipart/form-data;boundary={}", boundary))))
+            },
+            DocumentOrBodyInit::Document(ref d) => {
```

Comment:
A document is not `Extractable`. This should be done in `fn Send()`, under step 4 (first half).